pkg/controller/container-runtime-config/container_runtime_config_bootstrap.go

```diff
@@ -0,0 +1,96 @@
+package containerruntimeconfig
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/version"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// RunContainerRuntimeBootstrap generates ignition configs at bootstrap
+func RunContainerRuntimeBootstrap(templateDir string, crconfigs []*mcfgv1.ContainerRuntimeConfig, controllerConfig *mcfgv1.ControllerConfig, mcpPools []*mcfgv1.MachineConfigPool) ([]*mcfgv1.MachineConfig, error) {
+	var res []*mcfgv1.MachineConfig
+	managedKeyExist := make(map[string]bool)
+	for _, cfg := range crconfigs {
+		// use selector since label matching part of a ContaineRuntimeConfig is not handled during the bootstrap
+		selector, err := metav1.LabelSelectorAsSelector(cfg.Spec.MachineConfigPoolSelector)
+		if err != nil {
+			return nil, fmt.Errorf("invalid label selector: %v", err)
+		}
+		for _, pool := range mcpPools {
+			// If a pool with a nil or empty selector creeps in, it should match nothing, not everything.
+			// skip the pool if no matched label for containerruntime config
+			if selector.Empty() || !selector.Matches(labels.Set(pool.Labels)) {
+				continue
+			}
+			role := pool.Name
+			// Generate the original ContainerRuntimeConfig
+			originalStorageIgn, _, _, err := generateOriginalContainerRuntimeConfigs(templateDir, controllerConfig, role)
+			if err != nil {
+				return nil, fmt.Errorf("could not generate origin ContainerRuntime Configs: %v", err)
+			}
+
+			var configFileList []generatedConfigFile
+			ctrcfg := cfg.Spec.ContainerRuntimeConfig
+			if !ctrcfg.OverlaySize.IsZero() {
+				storageTOML, err := mergeConfigChanges(originalStorageIgn, cfg, updateStorageConfig)
+				if err != nil {
+					glog.V(2).Infoln(cfg, err, "error merging user changes to storage.conf: %v", err)
+				} else {
+					configFileList = append(configFileList, generatedConfigFile{filePath: storageConfigPath, data: storageTOML})
+				}
+			}
+			// Create the cri-o drop-in files
+			if ctrcfg.LogLevel != "" || ctrcfg.PidsLimit != nil || ctrcfg.LogSizeMax != (resource.Quantity{}) {
```

Comment:
Should logSizeMax also compare with `IsZero()` for consistency?

Reply:
fixed